includes/model/Shifts_model.php

```diff
@@ -398,7 +398,7 @@ function Shift_free_entries(AngelType $needed_angeltype, $shift_entries)
 {
     $taken = 0;
     foreach ($shift_entries as $shift_entry) {
-        if (!$shift_entry->freeloaded) {
+        if (is_null($shift_entry->freeload_user_id)) {
```

Comment:
Nitpick:

You use `is_null($shift_entry->freeload_user_id)` in about 10 places. I would suggest adding a small utility method to the ShiftEntry model. Something like this:

```php
function ShiftEntry_isFreeloaded(ShiftEntry $shiftEntry) {
    return !is_null($shift_entry->freeload_user_id);
}
```

Reply:
done